src/main/scala/firrtl/Emitter.scala

```diff
@@ -1258,8 +1258,9 @@ class SystemVerilogEmitter extends VerilogEmitter {
 
   override def addFormalStatement(formals: mutable.Map[Expression, ArrayBuffer[Seq[Any]]],
                                   clk: Expression, en: Expression,
-                                  stmt: Seq[Any], info: Info): Unit = {
+                                  stmt: Seq[Any], info: Info, msg: StringLit): Unit = {
     val lines = formals.getOrElseUpdate(clk, ArrayBuffer[Seq[Any]]())
+    lines += Seq("// ", msg.serialize)
     lines += Seq("if (", en, ") begin")
```

Comment:
I'm about to add source locator information to always blocks so I'm wondering how the combination should look:

#### This PR

```verilog
always @(posedge clock) begin
  // assume input is 0xAA
  if (1'h1) begin
    assume(inputEquals0xAA);
  end
end
```

#### My idea for my PR

```verilog
always @(posedge clock) begin
  if (1'h1) begin // @[MyFile.scala 20:14]
    assume(inputEquals0xAA); // @[MyFile.scala 21:16]
  end
end
```

#### Combo?

```verilog
always @(posedge clock) begin
  // assume input is 0xAA
  if (1'h1) begin // @[MyFile.scala 20:14]
    assume(inputEquals0xAA); // @[MyFile.scala 21:16]
  end
end
```

Reply:
Also should the message comment maybe go inside the if?

Reply:
> I'm about to add source locator information to always blocks so I'm wondering how the combination should look:

I think your suggested combo looks good.



> Also should the message comment maybe go inside the if?

I don't think so since the `en` signal which feeds into the `if` is part of the assertion.

Reply:
Oh fair, then what should I do with the source locator? I was assuming that was an `if` from a `when` in my example. I think it is more obvious to the user to put it on the `assume` itself but I don't feel that strongly

Reply:
> Oh fair, then what should I do with the source locator?

Is it excessive to just add it to both lines? This way people won't complain about missing source locators.